### PR TITLE
WIP: Document config.yaml

### PR DESCRIPTION
**Pull request by davidnuzik**
#### Proposed Changes ####

- Initial version of config_file.md added to the /docs/install portion of docs

#### Types of Changes ####

Docs

#### Verification ####

Review via docs site: https://docs.rke2.io/ in the Install section

#### Linked Issues ####

Addresses https://github.com/rancher/rke2/issues/450

### CODE REVIEW

File: docs/install/config_file.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: Configuration File
+---
+
+At runtime, RKE2 will check for the presence of `/etc/rancher/rke2/config.yaml`. The config.yaml file can be utilized to specify CLI arguments.
+
+An example of a basic `server` config file is below:
+
+```yaml
+# config.yaml
```
Comment (briandowns) [inline docs/install/config_file.md:10]: Is this required in the output? Seems redundant.

File: docs/install/config_file.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  - "foo=bar"
+  - "something=amazing"
+```
+
+In general, CLI arguments map to their respective YAML key, with repeatable CLI arguments being represented as YAML lists.
+
+An identical configuration using solely CLI arguments is shown below to demonstrate this:
+
+```bash
+# bash
```
Comment (briandowns) [inline docs/install/config_file.md:24]: Is this required in the output? Seems redundant.

File: docs/install/quickstart.md [hunk 3]
```diff
@@ -75,4 +75,6 @@ systemctl start rke2-agent.service
 journalctl -u rke2-agent -f
 ```
 
-**Note:** Each machine must have a unique hostname. If your machines do not have unique hostnames, set the `node-name` parameter in the `config.yaml` file and provide a value with a valid and unique hostname for each node.
+**Note:** Each machine must have a unique hostname. If your machines do not have unique hostnames, set the `node-name` parameter in the `config.yaml` file and provide a value with a valid and unique hostname for each node
```
Comment (briandowns) [inline docs/install/quickstart.md:78]:
```suggestion
**Note:** Each machine must have a unique hostname. If your machines do not have unique hostnames, set the `node-name` parameter in the `config.yaml` file and provide a value with a valid and unique hostname for each node.
```

**Comment by davidnuzik** [comment]
This isn't ready for review yet
My bad I didn't know it would auto-assign reviewers.

**Comment by davidnuzik** [comment]
Closing out. Looks like I need to try to get something in similar to the k3s install options section we have, not just the config file. So I'm going to start fresh and get working on this and enter a new draft PR when I'm ready later.

**Pull request closed without merging**